okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java

```diff
@@ -1472,7 +1472,12 @@ static String percentDecode(String encoded, int pos, int limit) {
     }
 
     // Fast path: no characters in [pos..limit) required decoding.
-    return encoded.substring(pos, limit);
+    String result = encoded.substring(pos, limit);
+    // + become ' ' for queries
+    if(isQuery) {
+      result = result.replace('+', ' ');
```

Comment:
yuck. I don't like that we have to loop through the characters of the string a 3rd time. Can't `percentDecode` do this job too?